woocommerce/woocommerce-ios
# [SwiftUI Payments menu] Restore Spotlight search for Payments

Closes: #11342 


## Description


Before the SwiftUI payments menu rewrite, we used UIKit hooks to allow searchability of the Payments menu. Our registration code set the user activity as the ViewController’s user activity, which meant that the system handled `becomeCurrent` calls as appropriate.

Our workaround was to make a UIViewController that we only used for registering a user activity; however, since that was never displayed, it never became current, and the user activity wasn’t indexed.

This commit changes the workaround to directly call `becomeCurrent` on the user activity, so that it’s registered for search.

Additionally, we wait for the hub menu to be visible before presenting payments, to avoid it sometimes only getting as far as the hub menu.

## Testing instructions


1. Launch the app
2. Navigate to `Menu > Payments` to register with Spotlight
3. Navigate to another part of the app; editing an order is a challenging screen for deeplinking to the menu
4. Background the app, then search "Payments" in spotlight
5. Tap the Woo link which shows
6. Observe that you're taken to the Payments Menu

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/2513d42d-98a0-476c-b1c9-97c3d6c78bf6



---
